addons/mail/static/src/components/autocomplete_input_view/autocomplete_input_view.js

```diff
@@ -0,0 +1,78 @@
+/** @odoo-module **/
+
+import { useComponentToModel } from '@mail/component_hooks/use_component_to_model';
+import { registerMessagingComponent } from '@mail/utils/messaging_component';
+
+const { Component, onMounted, onWillUnmount } = owl;
+
+export class AutocompleteInputView extends Component {
+
+    /**
+     * @override
+     */
+    setup() {
+        super.setup();
+        useComponentToModel({ fieldName: 'component' });
+        onMounted(() => this._mounted());
+        onWillUnmount(() => this._willUnmount());
+    }
+
+    _mounted() {
+        if (!this.root.el) {
+            return;
+        }
+        if (this.autocompleteInputView.isFocusOnMount) {
+            this.root.el.focus();
+        }
+
+        let args = {
+            autoFocus: true,
+            select: (ev, ui) => {
+                if (this.autocompleteInputView) {
+                    this.autocompleteInputView.onSelect(ev, ui);
+                }
+            },
+            source: (req, res) => {
+                if (this.autocompleteInputView) {
+                    this.autocompleteInputView.onSource(req, res);
+                }
+            },
+            focus: ev => ev.preventDefault(),
+            html: this.autocompleteInputView.isHtml,
+        };
+
+        if (this.autocompleteInputView.customClass) {
+            args.classes = { 'ui-autocomplete': this.autocompleteInputView.customClass };
+        }
+
+        const autoCompleteElem = $(this.root.el).autocomplete(args);
+        // Resize the autocomplete dropdown options to handle the long strings
+        // By setting the width of dropdown based on the width of the input element.
+        autoCompleteElem.data("ui-autocomplete")._resizeMenu = function () {
```

Comment:
Note that if you want to respect the conventions used in Discuss, you must use single quotes for "technical strings".

Reply:
I dont know if this should be in the scope of this PR, including the `const` change previously cause those are old codes. But I will make changes accordingly first and see if it is being approved later.